language_detector/main.py

```diff
@@ -1,9 +1,24 @@
+# -*- coding: utf-8 -*-
+
 """This is the entry point of the program."""
 
 from .languages import LANGUAGES
+import string
+import re
 
 
 def detect_language(text, languages):
-    """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text = re.compile('[%s]' % re.escape(string.punctuation)).sub('',text)     #remove punctuation in string
```

Comment:
NIT: I suggest reading [this](http://stackoverflow.com/questions/452104/is-it-worth-using-pythons-re-compile). Most developers use `re.compile` for a reusing (and optimization _when_ necessary) regex, otherwise `re.sub` (in your case) should suffice.